src/mpid/ch4/netmod/ofi/ofi_events.c

```diff
@@ -653,7 +654,15 @@ static int am_recv_event(struct fi_cq_tagged_entry *wc, MPIR_Request * rreq)
 
         case MPIDI_AMTYPE_RDMA_READ:
             /* buffer always copied together (there is no payload, just LMT header) */
-            p_data = (char *) orig_buf + sizeof(*am_hdr) + am_hdr->am_hdr_sz;
+#if NEEDS_STRICT_ALIGNMENT
+            MPIDI_OFI_lmt_msg_payload_t temp_rdma_payload;
+            if (!repeat) {
+                memcpy(&temp_rdma_payload, (char *) orig_buf + sizeof(*am_hdr) + am_hdr->am_hdr_sz,
+                       sizeof(MPIDI_OFI_lmt_msg_payload_t));
+                p_data = (void *) &temp_rdma_payload;
+            } else
+#endif
+                p_data = (char *) orig_buf + sizeof(*am_hdr) + am_hdr->am_hdr_sz;
             mpi_errno = MPIDI_OFI_handle_rdma_read(am_hdr, am_hdr + 1,
```

Comment:
Can you please add a brackets for the else part so it is clear that it is within the else branch?
```
#endif
{
    p_data = (...
}
```